src/acp/control.test.ts

```diff
@@ -322,6 +331,707 @@ describe("removeAccessControl", () => {
+        DataFactory.namedNode("https://some.pod/policy-resource#policy")
+      )
+    );
+    const resourceWithAcr = addMockAcrTo(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      accessControlResource
+    );
+
+    const policyUrls = getMemberAcrPolicyUrlAll(resourceWithAcr);
+
+    expect(policyUrls).toEqual([]);
+  });
+});
+
+describe("removeAcrPolicyUrl", () => {
+  it("removes the given URL as a Policy from the given ACR", () => {
+    const accessControlResource = mockAcrFor("https://some.pod/resource");
+    accessControlResource.add(
+      DataFactory.quad(
+        DataFactory.namedNode(getSourceUrl(accessControlResource)),
+        DataFactory.namedNode(acp.access),
+        DataFactory.namedNode("https://some.pod/policy-resource#policy")
+      )
+    );
+    const resourceWithAcr = addMockAcrTo(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      accessControlResource
+    );
+
+    const updatedResourceWithAcr = removeAcrPolicyUrl(
+      resourceWithAcr,
+      "https://some.pod/policy-resource#policy"
+    );
+    const acrQuads = Array.from(updatedResourceWithAcr.internal_acp.acr);
+
+    expect(acrQuads).toEqual([]);
+  });
+
+  it("returns the input unchanged if there was nothing to remove", () => {
+    const accessControlResource = mockAcrFor("https://some.pod/resource");
+    const resourceWithAcr = addMockAcrTo(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      accessControlResource
+    );
+
+    const updatedResourceWithAcr = removeAcrPolicyUrl(
+      resourceWithAcr,
+      "https://some.pod/policy-resource#policy"
+    );
+
+    expect(updatedResourceWithAcr).toEqual(resourceWithAcr);
+  });
+
+  it("does not remove existing Policy URLs", () => {
+    const accessControlResource = mockAcrFor("https://some.pod/resource");
+    accessControlResource.add(
+      DataFactory.quad(
+        DataFactory.namedNode(getSourceUrl(accessControlResource)),
+        DataFactory.namedNode(acp.access),
+        DataFactory.namedNode("https://some.pod/policy-resource#policy")
+      )
+    );
+    accessControlResource.add(
+      DataFactory.quad(
+        DataFactory.namedNode(getSourceUrl(accessControlResource)),
+        DataFactory.namedNode(acp.access),
+        DataFactory.namedNode("https://some.pod/policy-resource#other-policy")
+      )
+    );
+    const resourceWithAcr = addMockAcrTo(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      accessControlResource
+    );
+
+    const updatedResourceWithAcr = removeAcrPolicyUrl(
+      resourceWithAcr,
+      "https://some.pod/policy-resource#policy"
+    );
+    const acrQuads = Array.from(updatedResourceWithAcr.internal_acp.acr);
+
+    expect(acrQuads).toHaveLength(1);
+    expect(acrQuads[0].predicate.value).toBe(acp.access);
+    expect(acrQuads[0].object.value).toBe(
+      "https://some.pod/policy-resource#other-policy"
+    );
+  });
+
+  it("does not remove Member Control Policy URLs", () => {
```

Comment:
I think here the test could verify that the member ACR policy isn't removed even if it has a matching IRI, otherwise it's a bit redundant with the previous test.

Reply:
Oh wow, good catch - you're really thorough. Here: 610d2f691753ef30ace7adc47b1885706d0b0e6d